src/Vehicle/Vehicle.cc

```diff
@@ -3264,12 +3275,21 @@ void Vehicle::_handleCommandAck(mavlink_message_t& message)
 #endif
 
     int entryIndex = _findMavCommandListEntryIndex(message.compid, static_cast<MAV_CMD>(ack.command));
-    bool commandInList = false;
     if (entryIndex != -1) {
-        MavCommandListEntry_t commandEntry = _mavCommandList.takeAt(entryIndex);
-        if (commandEntry.command == ack.command) {
-            if (commandEntry.resultHandler) {
-                (*commandEntry.resultHandler)(commandEntry.resultHandlerData, message.compid, static_cast<MAV_RESULT>(ack.result), ack.progress, MavCmdResultCommandResultOnly);
+        if (ack.result == MAV_RESULT_IN_PROGRESS) {
+            MavCommandListEntry_t commandEntry = _mavCommandList.at(entryIndex);  // Command has not completed yet, don't remove
+
+            commandEntry.maxTries = 1;              // Vehicle responsed to command so don't retry
+            commandEntry.elapsedTimer.restart();    // We've heard from vehicle, restart elapsed timer for no ack received timeout
```

Comment:
Nice, restarting the timer is important!

Reply:
Can we also agree that restarting the actual timer is more important? Here a copy of the original elapsedTimer is restarted. Because [this](https://github.com/mavlink/qgroundcontrol/blob/39fcc8299ed9824bc972802ec7e678540d374b3d/src/Vehicle/Vehicle.cc#L3280) is a copy.